exercises/practice/diamond/diamond_test.go

```diff
@@ -218,3 +219,16 @@ func TestCharOutOfRangeShouldGiveError(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestDiamond(t *testing.T) {
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			expected := fmt.Sprintf("%s\n", strings.Join(testCase.expected, "\n"))
+			got, _ := Gen([]byte(testCase.input)[0])
+
+			if got != expected {
+				t.Fatalf("Gen(%q): %q\n\t Expected: %q\n\t Got: %q", testCase.input, testCase.description, expected, got)
```

Comment:
```suggestion
				t.Fatalf("Gen(%q)\nExpected:\n%s\n(len=%d)\nGot:\n%s\n(len=%d)",
					testCase.input,
					expected, len(expected),
					got, len(got))
```



- Removed the test description that is redundant, as Go will print the subtest names next to each test.
- Did some testing and came up with this string to make the output look nice:

<img width="634" alt="image" src="https://user-images.githubusercontent.com/6928620/174479208-358aae3c-f744-4325-baaa-90236062b81f.png">

`%s` ended up being better here just so the `\n` are properly interpreted. The problem with using `%s` is that makes whitespace differences hard to see, so I also included the length of the string in the output.